sql/src/main/java/io/crate/action/sql/query/FieldsVisitorInputFieldComparator.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to CRATE.IO GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.action.sql.query;
+
+import com.google.common.base.Throwables;
+import io.crate.operation.Input;
+import io.crate.operation.reference.doc.lucene.CollectorContext;
+import io.crate.operation.reference.doc.lucene.LuceneCollectorExpression;
+import io.crate.types.DataType;
+import org.apache.lucene.index.AtomicReaderContext;
+import org.apache.lucene.index.IndexReader;
+import org.apache.lucene.search.FieldComparator;
+
+import java.io.IOException;
+import java.util.List;
+
+public class FieldsVisitorInputFieldComparator extends InputFieldComparator {
+
+    private final CollectorContext collectorContext;
+    private IndexReader currentReader;
+
+
+    public FieldsVisitorInputFieldComparator(int numHits,
```

Comment:
do you need the collectorcontext here? would probably make sense to pass the visitor in the constructor